mod_reforged/hooks/skills/effects/dazed_effect.nut

```diff
@@ -0,0 +1,51 @@
+::mods_hookExactClass("skills/effects/dazed_effect", function(o) {
+	o.m.TwoTurnColor <- ::createColor("#ff6600");	// Orange
+
+	local onUpdate = o.onUpdate;
+	o.onUpdate = function( _properties )
+	{
+		onUpdate(_properties);
+		this.updateSprite();
+	}
+
+// New Functions
+	// Initialize the TurnsLeft with a new value. Will always set the Turns to atleast 1. This function already exists for stunned_effect
+	o.setTurns <- function( _amount )
+	{
+		this.m.TurnsLeft = _amount;
+		if (this.getContainer() != null)
+		{
+			this.m.TurnsLeft += this.getContainer().getActor().getCurrentProperties().NegativeStatusEffectDuration;
```

Comment:
This is a simple operation, but the code is too long. I suggest the following two options (I prefer the ternary):
```squirrel
if (this.getContainer() == null)
{
   this.m.TurnsLeft = _amount
}
else
{
   `this.m.TurnsLeft = ::Math.max(1, _amount + this.getContainer().getActor().getCurrentProperties().NegativeStatusEffectDuration)`
}
this.updateSprite();
```
I would've used a ternary, but it's up to you:
```squirrel
this.m.TurnsLeft = this.getContainer() == null ? _amount : ::Math.max(1, _amount + this.getContainer().getActor().getCurrentProperties().NegativeStatusEffectDuration);
this.updateSprite();
```

Reply:
Your solution has less but longer lines. I don't see where this would be a clear improvement. Just seems like a side-grade depending on preference.
In this case the function body is already very small so I prefer using a few extra lines for each step.

Your suggestion is missing a `::Math.max(1, _amount)` call for the path where getContainer is null